If statement optimization

Tattersalt-Kanzaki opened the pull request:
Got rid of unnecessary checks in if statements. These conditions are always true, as they are checked at a higher nesting level.

Tattersalt-Kanzaki committed:
got rid of expressions that are always true

brechtsanders commented:
Why would you get rid of these extra checks?
They are intended to catch function calls with invalid values.

Tattersalt-Kanzaki commented:
Hi!

1. I removed "data->callback" check  on line 767 xlsxio/lib/xlsxio_read.c because it is already checked on line 762 of same file;

2.  I removed "data->colnr > 0" check on line 1145 xlsxio/lib/xlsxio_read.c because it is already checked on line 1136 of same file;

3. I removed "!handle->sheetopen" check on line 918 xlsxio/lib/xlsxio_write.c because it is already checked on line 901 of same file.

Tattersalt-Kanzaki commented:
File xlsxio/lib/xlsxio_read.c
![1](https://github.com/user-attachments/assets/fd72c658-aa74-4748-bda5-8aad1c22a091)

As we see here statement (data->callback != 0) is already checked, so we can get rid of "if" statement on line 767.

Tattersalt-Kanzaki commented:
File xlsxio/lib/xlsxio_read.c
![2](https://github.com/user-attachments/assets/3ce3727d-3c12-4d81-b975-bd542bc359be)

We can get rid of check data->color > 0 (line 1145) because data->color was assigned from cellcolnr (line 1139) which is positive (check on line 1136). Also data->colnr can't be negative because it is declared as size_t;

Tattersalt-Kanzaki commented:
File xlsxio/lib/xlsxio_write.c
![3](https://github.com/user-attachments/assets/6a14b08c-d2cc-4196-82fe-8c14a9e39f30)

We can get rid of check !handle->sheetopen (line 918) because this check is happening in else statement of check handle->sheetopen (line 901).

Tattersalt-Kanzaki commented:
@brechtsanders,  I hope I've managed to explain my pull request.
If something is wrong, please let me know.
